### PR TITLE
feat(example): add use-as-ts-module example

### LINKED ISSUE #350: Example: use as typescript module
We want to add typescript version of [use-as-module](https://github.com/textlint/textlint/tree/master/examples/use-as-module "use-as-module").

It is useful for testing type definition.
We have added `types` to `textlint` module #348  and `@textlint/kernel`.

That exaple help to test the type defintion.

### PR DESCRIPTION
**Pull request by 0x6b**
Hi, I created TypeScript example for your review.

Closes #350.

### CODE REVIEW

**Comment by 0x6b** [comment]
Example should not break CI build. Why test on my local environment have passed. Really sorry for this.

File: examples/use-as-ts-module/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@textlint/*": [
+        "./packages/@textlint/*/src"
+      ],
+      "*": [
+        "./packages/*/src"
+      ]
+    },
```
Comment (azu) [inline examples/use-as-ts-module/tsconfig.json]: Probably, extends `tsconfig.base.json` resolve this issue?

CI error is sImilar to #414 ?

**Comment by 0x6b** [comment]
@azu, thank you for your help as always. Extending `tsconfig.base.json` seems to work on my local docker environment. I will double check before adding more commits into this pull request in a week.

**Comment by 0x6b** [comment]
@azu, as you suggested, extending `tsconfig.base.json` solved the issue. Thank you. Please review revised pull request.

As for #414 (TypeScript module resolver requires `index.ts`), we might have similar issue with `textlint-formatter`. I will open another issue and pull request.

**Comment by azu** [comment]
> as you suggested, extending tsconfig.base.json solved the issue. Thank you. Please review revised pull request.

OK. I will review at tonight.

**Comment by azu** [comment]
@0x6b Thanks!

**Pull request merged**